examples/desktop-notification/notification.go

```diff
@@ -1,22 +1,20 @@
+//go:generate gallium-main
 //go:generate go-bindata -o bindata.go gopher.png
 
 package main
 
 import (
 	"log"
 	"os"
-	"runtime"
 
 	"github.com/alexflint/gallium"
 )
 
-func main() {
-	runtime.LockOSThread()
-	gallium.RedirectStdoutStderr(os.ExpandEnv("$HOME/Library/Logs/Gallium.log"))
-	gallium.Loop(os.Args, onReady)
-}
+type notifications struct{}
```

Comment:
Developers could use their custom type to store application state or dependencies which they can initialise in `New()`.

The struct could also implement other gallium interfaces in the future. e.g.: how about being able to define a method that would return the default menu for the app?